guppylang/std/builtins.py

```diff
@@ -612,31 +623,38 @@ def __new__(): ...
 
     @guppy
     @no_type_check
-    def __iter__(self: array[L, n] @ owned) -> "SizedIter[ArrayIter[L, n], n]":
+    def __iter__(self: array[L, n] @ owned) -> SizedIter[ArrayIter[L, n], n]:
         return SizedIter(ArrayIter(self, 0))
 
     @guppy.custom(CopyInoutCompiler(), ArrayCopyChecker())
     def copy(self: array[T, n]) -> array[T, n]: ...
 
+    def __new__(cls, *args: _T) -> builtins.list[_T]:  # type: ignore[no-redef]  # noqa: F811
+        # Runtime array constructor that is used for comptime. We return an actual list
+        # in line with the comptime unpacking logic that turns arrays into lists.
```

Comment:
Just checking it shouldn't be "lists into arrays" rather than "arrays into lists"

Reply:
The point is that if a user writes ``array(x, y, z)``, they will actually get back a list in comptime mode. Similarly, if a function takes an array, the comptime implementation will actually get a list

Reply:
Ah I think I was mixing things up with `comptime` python lists being treated as static arrays, thanks